### PR TITLE
Update dependency and docs: jwt_verify_lib (add HS256 support)

### PR DESCRIPTION
**Pull request by ryanchapman**
Signed-off-by: Ryan A. Chapman <ryan@rchapman.org>

Description: update jwt_verify_lib to support HS256 tokens, also update documentation to show that Envoy now supports HS256 as well as RS384 and RS512 (see #8212)
Risk Level: low
Testing: upstream unit tests
Docs Changes: none
Release Notes: none (should there be some?)

### CODE REVIEW

**Comment by lizan** [comment]
Can you update the doc at https://github.com/envoyproxy/envoy/blob/master/docs/root/configuration/http/http_filters/jwt_authn_filter.rst ?

**Comment by ryanchapman** [comment]
@lizan I did that as well in this PR.  

```
Only ES256, HS256, RS256, RS384, and RS512 are supported for the JWT alg.
```

If you like to see more than that, please let me know.

**Comment by lizan** [comment]
Ah sorry I missed that, thought you didn't change doc from PR description.

File: docs/root/configuration/http/http_filters/jwt_authn_filter.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ This HTTP filter can be used to verify JSON Web Token (JWT). It will verify its
 JWKS is needed to verify JWT signatures. They can be specified in the filter config or can be fetched remotely from a JWKS server.
 
 .. attention::
-   Only ES256 and RS256 are supported for the JWT alg.
```
Comment (lizan) [inline docs/root/configuration/http/http_filters/jwt_authn_filter.rst]: nit: drop "Only", drop "," before "and".

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -160,10 +160,10 @@ REPOSITORY_LOCATIONS = dict(
         urls = ["https://github.com/msgpack/msgpack-c/releases/download/cpp-3.2.0/msgpack-3.2.0.tar.gz"],
     ),
     com_github_google_jwt_verify = dict(
-        sha256 = "2d57d336239d5fe36a03849ddbea1bff09a1720e1c4a46bbb9743c71732b0d43",
-        strip_prefix = "jwt_verify_lib-0f14d43f20381cfae0469cb2309b2e220c0f0ea3",
-        # 2019-07-08
-        urls = ["https://github.com/google/jwt_verify_lib/archive/0f14d43f20381cfae0469cb2309b2e220c0f0ea3.tar.gz"],
+        sha256 = "38a93926f362a330a2a4489ed799c260df0bc305417e2bb44d6745671d9641d7",
+        strip_prefix = "jwt_verify_lib-7e3191b0dcb72835aa63e308a53b541e7fda5458",
+        # 2019-09-23
```
[the remark this answers is not in the document]
Comment (ryanchapman) [reply bazel/repository_locations.bzl]: Fixed that.  Looks like checks are passing again :)

**Pull request merged**